config/routes.rb

```diff
@@ -1,6 +1,12 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  root "users#index"
+  get "users/:id", to: "users#user"
+  get "users/:id/posts", to: "posts#posts"
+  get "users/:id/posts/:id", to: "posts#post"
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show]
+  end
```

Comment:
#### Optional

- [optional ] 👏 Great job on setting up your routes! 🚀 However, the custom routes you've defined seem to overlap with the standard RESTful routes created by `resources. ` 🔄


#### Your custom routes

```


root "users#index"
  get "users/:id", to: "users#user"
  get "users/:id/posts", to: "posts#posts"
  get "users/:id/posts/:id", to: "posts#post"
```

#### Your standard routes



```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```


By removing the custom routes, you'll depend entirely on the RESTful routes offered by resources. This not only simplifies your code but also aligns more closely with RESTful conventions, making your application more predictable and easier to manage. 🛤️🧹